src/datasets/dataset_dict.py

```diff
@@ -1417,6 +1423,57 @@ def push_to_hub(
             revision=branch,
         )
 
+    def to_pandas(
+        self,
+        splits: Optional[Union[Literal["all"], List[str]]] = None,
+        batch_size: Optional[int] = None,
+        batched: bool = False,
+    ) -> Union[pd.DataFrame, Iterator[pd.DataFrame]]:
+        """Returns the dataset as a :class:`pandas.DataFrame`. Can also return a generator for large datasets.
+
+        You must specify which splits to convert if the dataset is made of multiple splits.
+
+        Args:
+            splits (:obj:`List[str]`, optional): List of splits to convert to a DataFrame.
+                You don't need to specify the splits if there's only one.
+                Use splits="all" to convert all the splits (they will be converted in the order of the dictionary).
+            batched (:obj:`bool`): Set to :obj:`True` to return a generator that yields the dataset as batches
+                of ``batch_size`` rows. Defaults to :obj:`False` (returns the whole datasets once)
+            batch_size (:obj:`int`, optional): The size (number of rows) of the batches if ``batched`` is `True`.
+                Defaults to :obj:`datasets.config.DEFAULT_MAX_BATCH_SIZE`.
+
+        Returns:
+            `pandas.DataFrame` or `Iterator[pandas.DataFrame]`
+
+        Example:
+
+        If the dataset has one split:
+        ```py
+        >>> df = dataset_dict.to_pandas()
+        ```
+
+        If the dataset has multiple splits:
+        ```py
+        >>> df_train = dataset_dict["train"].to_pandas()
+        >>> df_test = dataset_dict.to_pandas(splits="all")
+        >>> df_test = dataset_dict.to_pandas(splits=["train", "test"])
+        ```
+        """
+        self._check_values_type()
+        self._check_values_features()
+        if splits is None and len(self) > 1:
+            raise SplitsError(
+                "Failed to convert to pandas: please choose which splits to convert. "
+                f"Available splits: {list(self)}. For example:"
+                '\n    df = ds["train"].to_pandas()'
+                '\n    df = ds.to_pandas(splits=["train", "test"])'
+            )
+        splits = splits if splits is not None and splits != "all" else list(self)
+        if batched:
+            return (df for split in splits for df in self[split].to_pandas(batch_size=batch_size, batched=batched))
+        else:
+            return pd.concat([self[split].to_pandas() for split in splits])
+
```

Comment:
Maybe adding a validation to check if any of the passed split names is not in self, before raising an error when:
```python
self[split]
```